source/nodejs/adaptivecards/src/serialization.ts

```diff
@@ -1086,10 +1086,23 @@ export abstract class SerializableObject {
         }
     }
 
-    protected getValue(prop: PropertyDefinition): any {
-        return this._propertyBag.hasOwnProperty(prop.getInternalName())
-            ? this._propertyBag[prop.getInternalName()]
-            : prop.defaultValue;
+    protected getValue(prop: PropertyDefinition, shouldOverrideProperty: boolean = false): any {
+        const name = prop.getInternalName();
+        if (this._propertyBag.hasOwnProperty(name)) {
+            const propertyBagValue = this._propertyBag[name];
+
+            if (shouldOverrideProperty && this._rawProperties.hasOwnProperty(name)) {
+                const rawPropertyValue = this._rawProperties[name];
+
+                if (propertyBagValue !== rawPropertyValue && typeof rawPropertyValue === "string" && rawPropertyValue.includes("${")) {
```

Comment:
I haven't stepped through any of this code, so I don't really know, but is there any potential harm from this `.includes("${")` being overly broad?

consider when `rawPropertyValue === "\${I can put anything here}"` (note the `\`)

Reply:
I actually had another question regarding this piece of code.

Joseph suggested that I simplify the method to always get the value from `_rawProperties` if `shouldOverrideProperty` and remove the additional checks here.

That seems to be working fine, but it made me wonder why the `string` values are being accepted by the property `toJSON` methods.

For instance, why doesn't `BoolProperty` throw an error here when `value` is a `string`?
https://github.com/microsoft/AdaptiveCards/blob/4e987c5fbdd9725f9b769efc1c62af022a14e945/source/nodejs/adaptivecards/src/serialization.ts#L432-L438